### PR TITLE
Second Qm rounding fix

### PR DESCRIPTION
**Pull request by daschuer**
Here is another rounding fix. 

It works really great now. All my test sine files are now only at the center bin, as expected.
This was not the case before.  
Almost all tracks are always detected on a center bin as well.  Exceptions are passages with vocal and drum solos, which is IMHO the expected result. 

I have expetional tracks though. One of these interesting tracks is: 
Moby - Bodyrock (Olav Basoskis Da Hot Funk Da Freak Funk Remix)
https://www.youtube.com/watch?v=oDJ35h_0xWs

It is detected as G Major. Played together with an other G Major track is sounds badly off pitch. So I guess it is. See below. With this fixed algorithm, we have a chance to evaluate the quality of the detection and remove the key value from such a track. I like the idea, instead of consider it compatible to any other track. In the case of this Moby track I would brefere to make it as off pitch. 
What do you think?

```
fractional key pre-sorting: 2.33333
fractional key pre-sorting: 22
fractional key pre-sorting: 10
fractional key pre-sorting: 8
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 8
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 20
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 13.3333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 20.3333
fractional key pre-sorting: 2.33333
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 24
fractional key pre-sorting: 19
fractional key pre-sorting: 19
fractional key pre-sorting: 19
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 18.3333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 9.33333
fractional key pre-sorting: 18.3333
fractional key pre-sorting: 9
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 11
fractional key pre-sorting: 11
fractional key pre-sorting: 11
fractional key pre-sorting: 6
fractional key pre-sorting: 6
fractional key pre-sorting: 22.3333
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 23
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 2
fractional key pre-sorting: 1.33333
fractional key pre-sorting: 1.33333
fractional key pre-sorting: 18
fractional key pre-sorting: 18
fractional key pre-sorting: 18
fractional key pre-sorting: 1.33333
fractional key pre-sorting: 23
fractional key pre-sorting: 11.3333
fractional key pre-sorting: 1.33333
fractional key pre-sorting: 1.33333
fractional key pre-sorting: 15.3333
fractional key pre-sorting: 15.3333
fractional key pre-sorting: 15.3333
fractional key pre-sorting: 15.3333
fractional key pre-sorting: 11.3333
fractional key pre-sorting: 10
fractional key pre-sorting: 10
fractional key pre-sorting: 10
fractional key pre-sorting: 10
fractional key pre-sorting: 10
fractional key pre-sorting: 10
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 3
fractional key pre-sorting: 24
fractional key pre-sorting: 24
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 3
fractional key pre-sorting: 8
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 8
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 18.6667
fractional key pre-sorting: 18.6667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 18.6667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 16.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 7.66667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 0.666667
fractional key pre-sorting: 0.666667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 21.6667
fractional key pre-sorting: 14.6667
fractional key pre-sorting: 0.666667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 23.6667
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 3
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 3
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 8
fractional key pre-sorting: 9.66667
fractional key pre-sorting: 8
fractional key pre-sorting: 3
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8
fractional key pre-sorting: 8.33333
fractional key pre-sorting: 8.33333
fractional key pre-sorting: 8
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 3
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 13.6667
fractional key pre-sorting: 20.3333
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 20.3333
fractional key pre-sorting: 20.3333
fractional key pre-sorting: 2.66667
fractional key pre-sorting: 13.3333
fractional key pre-sorting: 13.3333
fractional key pre-sorting: 13.3333
fractional key pre-sorting: 8
Debug [AnalyzerThread 0 #2]: SoundSourceMP3 - Retry decoding MP3 frame @ 16987392
Debug [AnalyzerThread 0 #2]: SoundSourceMP3 - Recoverable MP3 frame decoding error: lost synchronization
Debug [AnalyzerThread 0 #2]: Key Detection complete
Debug [AnalyzerThread 0 #2]: Key Histogram
Debug [AnalyzerThread 0 #2]: 2 : "D♭" 950272
Debug [AnalyzerThread 0 #2]: 3 : "D" 4.9152e+06
Debug [AnalyzerThread 0 #2]: 6 : "F" 65536
Debug [AnalyzerThread 0 #2]: 8 : "G" 5.12653e+06
Debug [AnalyzerThread 0 #2]: 9 : "A♭" 819200
Debug [AnalyzerThread 0 #2]: 10 : "A" 655360
Debug [AnalyzerThread 0 #2]: 11 : "B♭" 360448
Debug [AnalyzerThread 0 #2]: 15 : "Dm" 262144
Debug [AnalyzerThread 0 #2]: 17 : "Em" 2.75251e+06
Debug [AnalyzerThread 0 #2]: 19 : "F♯m" 294912
Debug [AnalyzerThread 0 #2]: 22 : "Am" 327680
Debug [AnalyzerThread 0 #2]: 23 : "B♭m" 131072
Debug [AnalyzerThread 0 #2]: 24 : "Bm" 294912
```

### CODE REVIEW

File: lib/qm-dsp/dsp/keydetection/GetKeyMode.cpp [hunk 1]
```diff
@@ -268,8 +268,10 @@ int GetKeyMode::process(double *PCMData)
     // m_Keys[1] is C center  1 / 3 + 1 = 1
     // m_Keys[4] is D center  4 / 3 + 1 = 2
     // '+ 1' because we number keys 1-24, not 0-23.
-    key = MathUtilities::getMax( m_Keys, 2* m_BPO, &dummy ) / 3 + 1;
+    int maxBin = MathUtilities::getMax( m_Keys, 2* m_BPO, &dummy );
+    key = maxBin / 3 + 1;
 
+//    std::cout << "fractional key pre-sorting: " << (maxBin + 2) / 3.0 << std::endl;
```
Comment (daschuer) [inline lib/qm-dsp/dsp/keydetection/GetKeyMode.cpp]: Uncomment this for the fractional key debug output.

**Review by uklotzde — COMMENTED** [review]
With those changes the key_rounding.patch becomes obsolete and should be deleted.

What's our strategy to synchronize these changes with upstream or has the development stalled and no future changes are expected?

**Comment by daschuer** [comment]
> With those changes the key_rounding.patch becomes obsolete and should be deleted.

Oh yes, I need to update the patch.

> What's our strategy to synchronize these changes with upstream or has the development stalled and no future changes are expected?

I have published the patch upstream, and on GitHub. I gut no response so far. There is also an other unmerged pending patch with no responds. I guess, we can considder the development stalled.

**Comment by Be-ing** [comment]
The last [upstream commit](https://github.com/c4dm/qm-dsp/commits/master) was on May 1 and there were more commits in February this year, so I wouldn't consider upstream development stalled yet. It has only been a few days since the [upstream PR](https://github.com/c4dm/qm-dsp/pull/2) was opened. In the meantime, keeping a patch in our repository is fine.

**Comment by uklotzde** [comment]
Last commit 2018-05-01, last year.

I don't see any signifcant progress since their last release in 2015:
https://code.soundsoftware.ac.uk/projects/qm-dsp/repository/revisions

**Comment by uklotzde** [comment]
Only build fixes. No new features, no bug fixes.

**Comment by daschuer** [comment]
Done.

**Comment by uklotzde** [comment]
LGTM. Thanks for working on this tricky DSP code!

If the changes are not merged upstream we might think about forking the entire qm-dsp stuff and delete all files that we don't need.

**Pull request merged**